apple/internal/codesigning_support.bzl

```diff
@@ -407,6 +406,26 @@ def _post_process_and_sign_archive_action(
     compression_requested = defines.bool_value(ctx, "apple.compress_ipa", False)
     should_compress = (ctx.var["COMPILATION_MODE"] == "opt") or compression_requested
 
+    # TODO(b/163217926): These are kept the same for the three different actions
+    # that could be run to ensure anything keying off these values continues to
+    # work. After some data is collected, the values likely can be revisited and
+    # changed.
+    mnemonic = "ProcessAndSign"
+    progress_message = "Processing and signing %s" % ctx.label.name
+
+    # If there is no work to be done, skip the processing/signing action, just
+    # copy the file over.
+    has_work = any([signing_command_lines, ipa_post_processor_path, should_compress])
+    if not has_work:
+        ctx.actions.run_shell(
+            inputs = [input_archive],
+            outputs = [output_archive],
+            mnemonic = mnemonic,
+            progress_message = progress_message,
+            command = "cp -p '%s' '%s'" % (input_archive.path, output_archive.path),
```

Comment:
should this use `arguments = [input_archive, output_archive]` instead of embedding them into the command string?

Reply:
@thomasvl is there a reason you prefer one or the other?

Reply:
Not really.  Something short like this is usually a little clearer to me in string form.  I thinks I also tend to use string form when I use `run_shell` just so help make it clear shell rules will apply to the command.  When it's a straight `run`, I think we tend to build up the args more because they are complex and the separate arg makes it easier.